tests/framework/microvm.py

```diff
@@ -399,3 +400,15 @@ def start(self):
         """
         response = self.actions.put(action_type='InstanceStart')
         assert self._api_session.is_good_response(response.status_code)
+
+    def start_async(self):
+        """Start the microvm without blocking.
+        """
+        asyncio.ensure_future(self._start_async())
```

Comment:
No, `ensure_future` schedule the task (such as `create_task`) but from any awaitable object (not only coroutines) on the current event loop. Btw asyncio itself has nothing related to multi thread or such things, it's just an event loop.

Reply:
As @ProPheT777 said its not a separate thread its cooperative scheduling. Python doesn't get any performance advantage out of threads because of the [GIL](https://realpython.com/python-gil/) so async is the really the only way to go or spawning multiple processes.

My gut feel is that the tests may be better refactored to asyncrhonous. This may involve some changes to the client class as well.

This is more of a "fire and forget" function that you can use to presumably spawn multiple VMs RE the performance testing requirement.

If having all the tests properly async is something that's desirable I can have look at implementing that?